packages/synapse-react-client/src/utils/hooks/useLogin.ts

```diff
@@ -1,15 +1,28 @@
 import { Dispatch, SetStateAction, useEffect, useState } from 'react'
 import {
+  ErrorResponseCode,
   LoginResponse,
   TwoFactorAuthErrorResponse,
   TwoFactorAuthLoginRequest,
   TwoFactorAuthOtpType,
+  TwoFactorAuthResetRequest,
 } from '@sage-bionetworks/synapse-types'
 import { AUTHENTICATION_RECEIPT_LOCALSTORAGE_KEY } from '../SynapseConstants'
 import { useMutation } from '@tanstack/react-query'
 import { SynapseClientError } from '../SynapseClientError'
 import SynapseClient from '../../synapse-client'
 import { ONE_TIME_PASSWORD_STEP } from '../../components'
+import { noop } from 'lodash-es'
+import { useResetTwoFactorAuth } from '../../synapse-queries'
+
+export type UseLoginOptions = {
+  sessionCallback?: () => void
+  twoFaErrorResponse?: TwoFactorAuthErrorResponse
+  /* If a twoFactorAuthError is encountered (including passed in the twoFactorAuthenticationRequired arg), this callback will be invoked */
+  onTwoFactorAuthRequired?: (
+    twoFaToken: Pick<TwoFactorAuthErrorResponse, 'twoFaToken' | 'userId'>,
+  ) => void
+}
```

Comment:
Change the hook argument to an object containing parameters instead of adding more optional parameters

Reply:
Rename to UseLoginProps?

Reply:
This function is not a component, so 'props` would not be accurate